follej/session03/mailroom.py

```diff
@@ -0,0 +1,115 @@
+#!/usr/bin/env python3
+from statistics import mean
+
+donors = [[]] * 5
+
+donors[0] = ['William Gates, III', [65000, 68888.40]]
+donors[1] = ['Mark Zuckerberg', [65000, 68888.40, 600000]]
+donors[2] = ['Jeff Bezos', [65000, 68888.40, 10000]]
+donors[3] = ['Paul Allen', [65000, 68888.40]]
+donors[4] = ['Jesse Follet', [0.65, 1.50]]
+
+
+def create_a_report():
+    indices = sort_donors()
+    print_a_report(indices)
+
+
+def print_a_report(indices):
+    header = 'Donor Name                | Total Given | Num Gifts | Average Gift'
+    top_line = '-' * len(header)
+    print(header)
+    print(top_line)
+    for i in indices:
+        print(donors[i][0].ljust(26), end='')
+        print("$%12.2f" % (sum(donors[i][1])), end='')
+        print("%12d  " % (len(donors[i][1])), end='')
+        print("$%12.2f" % (mean(donors[i][1])))
+    print()
+
+
+def sort_donors():
+    amounts = []
+    for i in range(len(donors)):
+        amounts.append([sum(donors[i][1]), i])
+    sort_amounts = sorted(amounts, reverse=True)
+    sort_indices = []
+    for i in range(len(sort_amounts)):
+        sort_indices.append(sort_amounts[i][1])
+    return sort_indices
+
```

Comment:
Can you convert the two `for` loops above to list comps?  If you do, which do you like better for readability?